app/services/foreman_ansible/override_resolver.rb

```diff
@@ -0,0 +1,68 @@
+module ForemanAnsible
+  # Service which resolves override values for hosts and hostgroups
+  class OverrideResolver
+    attr_reader :overrides
+
+    def initialize(resource_id, resource_name, parent_id, variables)
```

Comment:
why do we initialize with id and resource name, can't we get the model instead?
We could even have a inner factory for Host/Hostgroup, that would take care of the actual resolving.

Reply:
We have [a method](https://github.com/theforeman/foreman_ansible/pull/313/files#diff-58ce55128ecc828dc6de7ef3ee290f1c107020c00d6ec20137386a51cb948070R26) that resolves it. Would you like me to move it to a different place?

Reply:
Well the whole premise of resolving variables only based on resource_name and parent is IMHO wrong, so I'd like to have directly the object, I'm explaining why in the review comment.